src/PostList.php

```diff
@@ -47,4 +52,53 @@ public function count()
     {
         return \count($this->data);
     }
+
+    /**
+     * Merge lists of posts.
+     *
+     * A new post list is returned, with:
+     *  - Duplicates removed, based on post title.
+     *  - Sorted by the number of hits in descending order.
+     *
+     * @param self ...$lists The other post lists to merge.
+     *
+     * @return \Lullabot\Parsely\PostList
+     */
+    public static function merge(self ...$lists): self
+    {
+        $merged = [];
+        foreach ($lists as $list) {
+            $merged = array_merge($merged, $list->getData());
+        }
+
+        // Remove duplicate results from the other list, based on title.
+        $titles = [];
+        $merged = array_filter($merged, function (Post $post) use (&$titles) {
+            if (\in_array($post->getTitle(), $titles)) {
+                return false;
+            }
+
+            $titles[] = $post->getTitle();
+
+            return true;
+        });
+
+        // Sort by the number of hits.
+        usort($merged, function (Post $a, Post $b) {
+            return $a->getHits() > $b->getHits() ? -1 : 1;
+        });
+
+        $new = new self();
+        $new->setData($merged);
+
+        return $new;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIterator()
```

Comment:
We need this so we can foreach() over a list of posts, and it's simpler than directly implementing \Iterator.